### PR TITLE
feat(KFLUXUI-1336): update merge request flow to accomodate PacState

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://redhat.atlassian.net/issues?filter=-1&selectedIssue=KFLUXUI-1336

## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

## Bug Fixes
* Improved PAC status accuracy by combining build-status details with pipeline results.
* Components now correctly reach “ready” when completion is reported, including cases without additional pagination.
* Prevented completed states from incorrectly appearing as pending or loading.
* Refined handling for migrated components, PUSH events, and bot-associated pull requests.

## Tests
* Expanded coverage for migration paths, pagination edge cases, event types, and build-status mappings.
* Added validation for enabled, disabled, configuring, pending, ready, and error states.

### CODE REVIEW

**Review by sahil143 — CHANGES_REQUESTED** [review]
Please update the PR description.

File: src/hooks/usePACStatesForComponents.ts [hunk 1]
```diff
@@ -120,7 +123,11 @@ const usePACStatesForComponents = (components: ComponentKind[]): PacStatesForCom
           (r) =>
             !r.metadata?.annotations?.[PipelineRunLabel.COMMIT_USER_LABEL]?.includes(prBotName),
         );
-        if (prMerged) {
+        if (
+          prMerged ||
+          (buildStatus?.message === PAC_STATE_DONE_MESSAGE &&
+            buildStatus?.pac?.state === ComponentBuildState.enabled)
```
Comment (rrosatti) [inline src/hooks/usePACStatesForComponents.ts]: how about extracting this check logic to a utility function, so it can be used on both `src/hooks/usePACStatesForComponents.ts` and `src/hooks/usePACState.ts`? 🤔 

example:
```tsx
  const isPacStateReady = (
    prMerged: PipelineRunKind | undefined,
    buildStatus: ComponentBuildStatus | null,
  ): boolean => {
    return (
      !!prMerged ||
      (buildStatus?.message === PAC_STATE_DONE_MESSAGE &&
        buildStatus?.pac?.state === ComponentBuildState.enabled)
    );
  };
```

so if eventually in the future we need to update this logic, we can update it in only one place instead of 2 (or more) :)

**Review by janaki29 — DISMISSED** [review]

File: src/hooks/usePACStatesForComponents.ts (hunk 1 above)
Comment (abhinandan13jan) [reply src/hooks/usePACStatesForComponents.ts]: I agree. Updated

**Review by janaki29 — DISMISSED** [review]

**Comment by rrosatti** [comment]
the e2e tests are failing when trying to search/find the `Merge pull request` button: https://github.com/konflux-ci/konflux-ui/actions/runs/31387460052/job/93451105307?pr=1045#step:11:350

Screenshot 2026-08-10 at 15 49 46

could you please take a look? we need to check whether it's something to be adjusted on the test itself or on the logic added/updated on this PR

**Review by rrosatti — COMMENTED** [review]
I just created a new component and opened the Component Details page:

Image

shouldn't the button here be `"Merge pull request"` instead of `"Custom"`? 🤔

**Comment by abhinandan13jan** [comment]
> I just created a new component and opened the Component Details page:
> 
> Image
> shouldn't the button here be `"Merge pull request"` instead of `"Custom"`? 🤔

Is that a button? I guess its a label showing the build pipeline plan type which is custom. @rrosatti

**Comment by rrosatti** [comment]
> Is that a button? I guess its a label showing the build pipeline plan type which is custom. @rrosatti

it's a Label which has an `onClick` action that opens the "Manage build pipelines" modal. That's how it currently works on `main` branch:

https://github.com/user-attachments/assets/a3a6f36b-41bc-4066-9e84-8b1fb4357be9

in the video above I'm using the same component from the previous screenshot. Notice that the label/button is "Merge pull request" and from the modal I can actually open GH and merge the PR :)

**Comment by abhinandan13jan** [comment]
After discussion along the original issue, it was advised to add  "Incoming" type to the list of PipelineRuns to be considered. It is already in `useLatestPushBuildPipelineRunForComponentV2`. So closing this out
